flink-core/src/main/java/org/apache/flink/api/common/externalresource/ExternalResourceInfo.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.api.common.externalresource;
+
+import org.apache.flink.annotation.PublicEvolving;
+
+import java.util.Collection;
+
+/**
+ * Contains the information of an external resource.
+ */
+@PublicEvolving
+public interface ExternalResourceInfo {
+
+	/**
+	 * Get the property indicated by the specified key.
+	 */
+	String getProperty(String key);
```

Comment:
I think it probably makes sense to return null here, just like `Map` or `Properties` in Java. However, I found the code style guide says:

> Always use Optional to return nullable values in the API/public methods except the case of a proven performance concern.

I'm not entirely sure about whether we should return `Optional` here.

Reply:
For this particular case, I think it also makes sense to return a Nullable value, since it's a common behavior for java key-value data structures.

I would like to hear opinions from other reviewers/committers on this.